tests/bench/coremark/coremark_main.c

```diff
@@ -466,21 +505,23 @@ for (i = 0; i < MULTITHREAD; i++)
 #endif
     }
 
+#if (MEM_METHOD == MEM_MALLOC)
+    for (i = 0; i < MULTITHREAD; i++)
+        portable_free(results[i].memblock[0]);
+#endif
+
+    /* And last call any target specific code for finalizing */
+    portable_fini(&(results[0].port));
 	if (total_errors == 0) {
 		ee_printf("Bench %s Success.\n", __func__);
 		return coremark_t_pass();
 	} else {
 		ee_printf("Bench %s Failed.\n", __func__);
 		return coremark_t_fail();
 	}
-#if (MEM_METHOD == MEM_MALLOC)
-    for (i = 0; i < MULTITHREAD; i++)
-        portable_free(results[i].memblock[0]);
-#endif
-    /* And last call any target specific code for finalizing */
-    portable_fini(&(results[0].port));
-
+#else
```

Comment:
Try to think about a way to keep main function less #ifdef/#endif pairs.

Reply:
fixed